accounts-db/src/tiered_storage.rs

```diff
@@ -49,6 +49,7 @@ pub struct TieredStorageFormat {
 #[derive(Debug)]
 pub struct TieredStorage {
     reader: OnceLock<TieredStorageReader>,
+    read_only: OnceLock<bool>,
```

Comment:
Sounds good.  I am also thinking about using the file size instead so that we can have a faster file_size() as well as checking whether is read-only.